publication/src/main/scala/hmda/publication/reports/disclosure/D51.scala

```diff
@@ -0,0 +1,155 @@
+package hmda.publication.reports.disclosure
+
+import java.util.Calendar
+
+import akka.NotUsed
+import akka.stream.scaladsl.Source
+import hmda.model.publication.reports.ApplicantIncomeEnum._
+import hmda.publication.reports._
+import hmda.model.publication.reports.ReportTypeEnum.Disclosure
+import hmda.model.publication.reports._
+import hmda.publication.reports.util.DateUtil._
+import hmda.publication.reports.util.DispositionType._
+import hmda.publication.reports.util.ReportUtil._
+import hmda.query.model.filing.LoanApplicationRegisterQuery
+
+import scala.concurrent.Future
+
+case class D51(
+  respondentId: String,
+  institutionName: String,
+  table: String,
+  reportType: ReportTypeEnum,
+  description: String,
+  year: Int,
+  reportDate: String,
+  msa: MSAReport,
+  applicantIncomes: List[ApplicantIncome],
+  total: List[Disposition]
```

Comment:
Are any of these fields going to be repeated across other reports?  We might be able to consolidate them and avoid a lot of repeated field declarations.